CIP-dapp-bridge-rework/README.md

```diff
@@ -0,0 +1,1169 @@
+The new approach still relies on code injection, but offers a more transparent API. The main idea is that a DApp should not perform the functions of a wallet.
+
+### What is it?
+
+It's automation, lowering the barrier to entry, separation of responsibilities, and scaling the complexity of development.
+
+### What is not?
+
+It's not "getting rid of CBOR", not transferring all functions to the wallet, not replacing everything with JSON.
+
+## Motivation: why is this CIP necessary?
+
+This CIP addresses several issues that complicate the development of the Cardano ecosystem with the current integration standard:
+
+1. **Complexity in Writing Client-Side Logic for DApps**: As they essentially need to implement their own wallet each time, including algorithms for coin selection, UTxO management, UTxO validation, TX building, etc. They shouldn't be doing this. There is already a separately written wallet that connects to their application for this purpose. <br><br>
+
+2. **Integration Difficulties**: This implies complicating the internal logic of wallets, adding endpoints for importing transactions, writing logic for checks/validation of foreign transactions, etc. This could be avoided if the transaction was created and processed natively by the wallet.<br><br>
+
+3. **Privacy Issues**: Currently, the wallet must pass all information to the DApp, including the UTxO set and full addresses on one hand, and trust the user's funds to the transaction building code inside the DApp on the other. This is not obvious to the user, and in case of problems, they will blame the wallet developers.<br><br>
+
+4. **Current Role of Wallets**: At present, the wallet is essentially just a synchronization point for the client side of the DApp with the blockchain. Yes, it generates user signatures, but, for example, if we use a HW-wallet, then the task of the SW-wallet is reduced to 'submit tx'. This is an incorrect design of communication.<br><br>
+
+5. **Scalable Complexity**: Currently, the integration standard implies that the complexity of the solution on the DApp side **must be** unjustifiably high even for elementary things that do not require knowledge of CBOR, CDDL, or an understanding of the difference between bech32 and base58.<br><br>
+
+6. **Reducing Dependencies**: Currently, all DApps developers **must** operate only with CBOR-CDDL entities, making them dependent on third-party libraries such as `cardano_serialization_lib`, and forcing them to integrate these libraries into their projects. This complicates the logic, increases the size of the loaded page, necessitates keeping up with updates, etc., thereby increasing the number of failure factors in the application's logic.<br><br>
```

Comment:
This CIP still uses CBOR and does not provide an abstraction from data layout because of that

Reply:
Yes. [This CIP is not about "getting rid of CBOR".](https://github.com/Fell-x27/CIPs/blob/c3f2dfd84447176403458ac0cbd810ed6f44f8d7/CIP-dapp-bridge-rework/README.md?plain=1#L77-L79). Actually we have to use CBOR for some cases and it's okay. 

This CIP is about **complexity reducing and automation**. So, if you don't need a CBOR, don't use it. If you want to produce some signatures, or finalize a transaction by yourself, then you are already working with CBOR data and it's not a problem for you.  But you still don't have to write your own "semi-wallet" to get your DApp working.

If you're already working with CBOR on the DApp side, trying to convert it to JSON in some way will only add new layers of complexity, instead of solving any problems.